### PR TITLE
Fix gradient jump indicator for FV

### LINKED ISSUE #24792: Mesh adaptivity not suppoerted for FV problems with variables restricted to subdomains
## Bug Description
When running a finite volume problem with a variable constrained to a subdomain, mesh adaptivity does not seem to be supported.

## Steps to Reproduce
Run the attached adaptivity_ex_FV.txt. An assertion is hit:

`Assertion `this->hasBlocks(elem->subdomain_id())' failed
The variable should be defined on the element's subdomain! This typically occurs when the user wants to evaluate the elements right next to the boundary of two variables (block boundary). The subdomain which is queried: 0 the subdomain of the element 1
at /Users/behnpa/projects/moose/framework/src/variables/MooseVariableFV.C, line 459`

A similar working example is provided for finite element in adaptivity_ex_FE.txt.

## Impact
Mesh adaptivity cannot be used for FV problems with variables restricted to subdomains.

[adaptivity_ex_FE.txt](https://github.com/idaholab/moose/files/11837957/adaptivity_ex_FE.txt)
[adaptivity_ex_FV.txt](https://github.com/idaholab/moose/files/11837958/adaptivity_ex_FV.txt)

### PR DESCRIPTION
**Pull request by GiudGiud**
And add some exception checking to a MG that was used to demonstrate the problem

closes #24792

### CODE REVIEW

File: test/tests/meshgenerators/refine_block_generator/tests [hunk 1]
```diff
@@ -31,11 +31,22 @@
...
+      cli_args = 'Mesh/refine/block="missing"'
+      expect_err = "The block 'missing' was not found within the mesh"
+      detail = "attempting to refine a block that does not exist in the mesh, and "
+    []
+    [wrong_size_inputs]
+      type = RunException
+      input = test_single.i
+      cli_args = 'Mesh/refine/refinement="1 2" --mesh-only'
+      expect_err = "The blocks and refinement parameter vectors should be the same size"
+      detail = "specifying a vector of refinement levels that does not match the vector of "
```
Comment (GiudGiud) [inline test/tests/meshgenerators/refine_block_generator/tests:48]:
```suggestion
      detail = "if specifying a vector of refinement levels that does not match the vector of "
```

File: test/tests/meshgenerators/refine_sideset_generator/tests [hunk 2]
```diff
@@ -29,12 +29,23 @@
...
+      cli_args = "Mesh/refine/boundaries='foo' --mesh-only"
+      expect_err = "The boundary *"
+      requirement = 'the specified boundary to be refined does not exist and'
+    []
+    [wrong_size_inputs]
+      type = RunException
+      input = test_left.i
+      cli_args = 'Mesh/refine/refinement="1 2" --mesh-only'
+      expect_err = "The boundaries and refinement parameter vectors should be the same size"
+      detail = "specifying a vector of refinement levels that does not match the vector of "
```
Comment (GiudGiud) [inline test/tests/meshgenerators/refine_sideset_generator/tests:47]:
```suggestion
      detail = "if specifying a vector of refinement levels that does not match the vector of "
```

File: framework/src/meshgenerators/RefineSidesetGenerator.C [hunk 3]
```diff
@@ -30,7 +30,8 @@ RefineSidesetGenerator::validParams()
   params.addParam<bool>(
       "enable_neighbor_refinement",
       true,
-      "Toggles whether neighboring level one elements should be refined or not. Defaults to true");
+      "Toggles whether neighboring level one elements should be refined or not. Defaults to true."
```
Comment (GiudGiud) [inline framework/src/meshgenerators/RefineSidesetGenerator.C:33]:
```suggestion
      "Toggles whether neighboring level one elements should be refined or not. Defaults to true. "
```

File: framework/src/meshgenerators/RefineBlockGenerator.C [hunk 4]
```diff
@@ -30,7 +30,8 @@ RefineBlockGenerator::validParams()
   params.addParam<bool>(
       "enable_neighbor_refinement",
       true,
-      "Toggles whether neighboring level one elements should be refined or not. Defaults to true");
+      "Toggles whether neighboring level one elements should be refined or not. Defaults to true."
```
Comment (GiudGiud) [inline framework/src/meshgenerators/RefineBlockGenerator.C:33]:
```suggestion
      "Toggles whether neighboring level one elements should be refined or not. Defaults to true. "
```

**Comment by GiudGiud** [comment]
dont really understand these test failures. how are my requirements not strings

**Review by lindsayad — APPROVED** [review]

**Pull request merged**